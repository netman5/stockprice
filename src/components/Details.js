/* eslint-disable arrow-body-style */
import React from 'react';

import Topbar from './Pages/Topbar';
import Card from './views/Card';

const Details = () => {
  return (
    <div>
      <Topbar />
      <Card />
    </div>
  );
};

export default Details;
