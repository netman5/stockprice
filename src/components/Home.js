import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { BallTriangle } from 'react-loader-spinner';
import { getStocksFromAPI } from '../redux/stocks/stocks';

const Home = () => {
  const dispatch = useDispatch();
  const state = useSelector((state) => state.stocks);
  const { data: stocks, loading } = state;

  useEffect(() => {
    if (stocks.length === 0) {
      dispatch(getStocksFromAPI());
    }
  }, []);

  return (
    <div>
      <h1>Hello</h1>
      <p>lorem</p>
      {loading ? (
        <div
          style={{
            width: '100%',
            height: '100',
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
          }}
        >
          {' '}
          <BallTriangle color="#2BAD60" height="80" width="80" timeout={300} />
        </div>
      ) : (
        <ul className="test">
          {stocks.map((item) => (
            <li key={item.symbol}>
              <Link to="details">{item.symbol}</Link>
            </li>
          ))}
        </ul>
      )}
    </div>
  );
};

export default Home;
