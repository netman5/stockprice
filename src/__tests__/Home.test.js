import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Routes } from 'react-router-dom';
import store from '../redux/configureStore';
import Home from '../components/Home';
import Details from '../components/Details';

test('if Home renders correctly', () => {
  const tree = renderer
    .create(
      <Routes>
        <Provider store={store}>
          <Home />
        </Provider>
      </Routes>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
//
test('Details page', () => {
  const tree = renderer
    .create(
      <Routes>
        <Provider store={store}>
          <Details />
        </Provider>
      </Routes>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
