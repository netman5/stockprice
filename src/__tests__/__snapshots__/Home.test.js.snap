// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="header"
    >
      <svg
        fill="currentColor"
        height={25}
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "white",
            "cursor": "pointer",
            "float": "left",
            "marginTop": "0.3rem",
          }
        }
        viewBox="0 0 1024 1024"
        width={25}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 0 0 0 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
        />
      </svg>
      <h1>
        Company Profile
      </h1>
    </div>
    <div
      className="icons"
    >
      <svg
        fill="currentColor"
        height={20}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512 624c93.9 0 170-75.2 170-168V232c0-92.8-76.1-168-170-168s-170 75.2-170 168v224c0 92.8 76.1 168 170 168zm330-170c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 140.3-113.7 254-254 254S258 594.3 258 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 168.7 126.6 307.9 290 327.6V884H326.7c-13.7 0-24.7 14.3-24.7 32v36c0 4.4 2.8 8 6.2 8h407.6c3.4 0 6.2-3.6 6.2-8v-36c0-17.7-11-32-24.7-32H548V782.1c165.3-18 294-158 294-328.1z"
        />
      </svg>
      <svg
        fill="currentColor"
        height={20}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M413.967 276.8c1.06-6.235 1.06-13.518 1.06-20.8s-1.06-13.518-1.06-20.8l44.667-34.318c4.26-3.118 5.319-8.317 2.13-13.518L418.215 115.6c-2.129-4.164-8.507-6.235-12.767-4.164l-53.186 20.801c-10.638-8.318-23.394-15.601-36.16-20.801l-7.448-55.117c-1.06-4.154-5.319-8.318-10.638-8.318h-85.098c-5.318 0-9.577 4.164-10.637 8.318l-8.508 55.117c-12.767 5.2-24.464 12.482-36.171 20.801l-53.186-20.801c-5.319-2.071-10.638 0-12.767 4.164L49.1 187.365c-2.119 4.153-1.061 10.399 2.129 13.518L96.97 235.2c0 7.282-1.06 13.518-1.06 20.8s1.06 13.518 1.06 20.8l-44.668 34.318c-4.26 3.118-5.318 8.317-2.13 13.518L92.721 396.4c2.13 4.164 8.508 6.235 12.767 4.164l53.187-20.801c10.637 8.318 23.394 15.601 36.16 20.801l8.508 55.117c1.069 5.2 5.318 8.318 10.637 8.318h85.098c5.319 0 9.578-4.164 10.638-8.318l8.518-55.117c12.757-5.2 24.464-12.482 36.16-20.801l53.187 20.801c5.318 2.071 10.637 0 12.767-4.164l42.549-71.765c2.129-4.153 1.06-10.399-2.13-13.518l-46.8-34.317zm-158.499 52c-41.489 0-74.46-32.235-74.46-72.8s32.971-72.8 74.46-72.8 74.461 32.235 74.461 72.8-32.972 72.8-74.461 72.8z"
        />
      </svg>
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "100",
        "justifyContent": "center",
        "margin": "10rem auto",
        "width": "100%",
      }
    }
  >
    <div
      data-testid="puff-loading"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <svg
        aria-label="puff-loading"
        data-testid="puff-svg"
        height="80"
        stroke="#2BAD60"
        viewBox="0 0 44 44"
        width="80"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
          strokeWidth="2"
        >
          <circle
            cx="22"
            cy="22"
            r={1}
          >
            <animate
              attributeName="r"
              begin="0s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.165, 0.84, 0.44, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 20"
            />
            <animate
              attributeName="strokeOpacity"
              begin="0s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.3, 0.61, 0.355, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 0"
            />
          </circle>
          <circle
            cx="22"
            cy="22"
            r={1}
          >
            <animate
              attributeName="r"
              begin="-0.9s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.165, 0.84, 0.44, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 20"
            />
            <animate
              attributeName="strokeOpacity"
              begin="-0.9s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.3, 0.61, 0.355, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 0"
            />
          </circle>
        </g>
      </svg>
    </div>
  </div>
</div>
`;

exports[`if Home renders correctly 1`] = `
<div
  className="app-container"
>
  <div>
    <h1>
      Random Top gainers stocks
    </h1>
    <div
      className="gainers"
      onClick={[Function]}
    >
      <svg
        className="gainers-link"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
      <div>
        <div
          className="symbol"
        >
          <p />
          <p />
        </div>
        <div
          className="price"
        >
          <p>
            Price: $undefined
          </p>
          <p>
            Change:
            <svg
              color="green"
              fill="none"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": "green",
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.0321 1.01712L7.75751 5.22761L9.161 6.65246L11.0197 4.82165L10.9644 22.9768L12.9644 22.9829L13.0195 4.86974L14.8177 6.69525L16.2425 5.29175L12.0321 1.01712Z"
                fill="currentColor"
              />
            </svg>
          </p>
        </div>
        <p
          className="percentage"
        >
          Percentage: 
          <svg
            color="green"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "green",
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.0321 1.01712L7.75751 5.22761L9.161 6.65246L11.0197 4.82165L10.9644 22.9768L12.9644 22.9829L13.0195 4.86974L14.8177 6.69525L16.2425 5.29175L12.0321 1.01712Z"
              fill="currentColor"
            />
          </svg>
           
          %
        </p>
      </div>
    </div>
  </div>
  <div
    className="input-container"
  >
    <label
      htmlFor="search"
    >
      <input
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="Search by symbol e.g, APPL"
        type="text"
        value=""
      />
    </label>
  </div>
  <div>
    <div
      className="main"
    >
      <h5>
        All Stocks
      </h5>
      <ul
        className="stock-category"
      />
    </div>
  </div>
</div>
`;
